tests/unit/test_bson_fields.py

```diff
@@ -39,6 +39,13 @@ class ModelWithDate(Model):
     assert inst.field.microsecond != sample_datetime.microsecond
     assert inst.field == sample_datetime.replace(microsecond=10000)
 
+    sample_datetime = sample_datetime.replace(microsecond=999501)
+    inst = ModelWithDate(field=sample_datetime)
+
+    assert inst.field == sample_datetime.replace(
+        microsecond=(sample_datetime.microsecond - sample_datetime.microsecond % 1000)
+    )
```

Comment:
Checking directly with the value would probably be more readable.
```suggestion
    assert inst.field == sample_datetime.replace(microsecond=999000)
```